fault-tolerance-test/src/restart.test.js

```diff
@@ -0,0 +1,269 @@
+const Docker = require('dockerode')
+const { sleep } = require('@jvalue/node-dry-basics')
+
+const { ADAPTER_URL, NOTIFICATION_URL, PIPELINE_URL, STORAGE_URL, STORAGE_MQ_URL, DOCKER_COMPOSE_FILE, ENV_FILE } = require('./util/env')
+const { DockerCompose, writeDockerLogs } = require('./util/docker-compose')
+const { waitForServicesToBeReady } = require('./util/waitForServices')
+const { publishEvent } = require('./util/amqp')
+const http = require('./util/http')
+
+const SECOND = 1000
+const MINUTE = 60 * SECOND
+const TEST_TIMEOUT = 5 * MINUTE
+
+const ALL_SERVICES = [
+  'adapter',
+  'adapter-db',
+  'adapter-outboxer',
+  'edge',
+  'notification',
+  'notification-db',
+  'pipeline',
+  'pipeline-db',
+  'pipeline-outboxer',
+  'rabbitmq',
+  'scheduler',
+  'storage',
+  'storage-db',
+  // 'storage-db-liquibase', //Only creates the database schema and then exits, therefore ignore
+  'storage-mq'
+].sort()
+
+const SERVICES_TO_KILL = [
+  ['adapter', initAdapter, testAdapter],
+  ['adapter-db', initAdapter, testAdapter],
+  ['adapter-outboxer', undefined, undefined],
+  ['notification', initNotification, testNotification],
+  ['notification-db', initNotification, testNotification],
+  ['pipeline', initPipeline, testPipeline],
+  ['pipeline-db', initPipeline, testPipeline],
+  ['pipeline-outboxer', undefined, undefined],
+  // rabbitmq has separate test method
+  ['scheduler', undefined, undefined],
+  ['storage', initStorage, testStorage],
+  ['storage-db', initStorage, testStorage],
+  ['storage-mq', initStorage, testStorage]
+]
+
+function getComposeName (container) {
+  return container.Labels['com.docker.compose.service']
+}
+
+describe('Restart test', () => {
+  let dockerCompose
+  let docker
+
+  beforeAll(() => {
+    dockerCompose = DockerCompose(DOCKER_COMPOSE_FILE, ENV_FILE)
+    docker = new Docker({})
+  })
+
+  beforeEach(async () => {
+    if ((await docker.listContainers()).length !== 0) {
+      throw Error('Can not execute restart test if other containers are running')
+    }
+
+    // Do not try to build the images because then timing is really hard
+    await dockerCompose('up -d --no-build adapter-db notification-db pipeline-db storage-db')
+    await dockerCompose('up -d --no-build storage-db-liquibase')
+    await dockerCompose('up -d --no-build adapter notification pipeline scheduler storage storage-mq')
```

Comment:
why not start all together except edge?

Reply:
From my experience, this is more reliable. When starting everything together, I experienced that sometimes the services timeout because the database initialization takes longer, because docker needs to launch all container at once.